### ISSUE TITLE
Featured product management

### ISSUE THREAD

**Issue by chhuovanna** [chhuovanna/secondhand#17]
# Before you create an issue make sure that:
- Your issue is **strictly related to the boilerplate** itself. Questions about Laravel in general belongs to the [laracasts](https://laracasts.com/discuss/) or [laravel](http://laravel.io/forum) forums.
- You have read the [wiki](https://github.com/rappasoft/laravel-5-boilerplate/wiki) thoroughly.
- You have searched for a similar issues among all the former issues (even closed ones).
- You have tried to replicate the issue with a clean install of the project.

# Be explicit
Try to be as explicit as you can when you ask anything.

# What version?
Please when it's necessary, provide the version of your copy of the boilerplate. You can find the version number in the original README.md.

**Comment by chhuovanna** [comment]
Here is the algo for add featured product
featured product
1. add featured product
    - add button for each product on the list :productcontroller@getproduct
    - add action listener (js) on the button in :productindex
        - promp the start_date (default=now) and end_date (empty for unlimited)
        - create ajax request send to url: product/featuredproduct/add, data:start_date, end_date, product_id
    - create route get product/featuredproduct/add associated with productcontroller@addFeaturedProduct :web.php
    - create function addFeaturedProduct:productcontroller
        - get start_date and end_date and product_id from request
        - check if there is any active featured_product for the product
            - yes: update
            - no : insert to table featured_product
                - if end_date is blank set it to 9999-01-01
                - if end_date < current date , insert fail

**Comment by chhuovanna** [comment]
sample code for modal
`<!DOCTYPE html>
<html lang="en">
<head>
  <title>Bootstrap Example</title>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/css/bootstrap.min.css">
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/js/bootstrap.min.js"></script>
</head>
<body>

<div class="container">
  <h2>Modal Example</h2>
  
  <button type="button" class="btn btn-info btn-lg" data-toggle="modal" data-target="#featured_product_modal">Open Modal</button>

  
  <div class="modal fade" id="featured_product_modal" role="dialog">
    <div class="modal-dialog">
    
      
      <div class="modal-content">
        <div class="modal-header">
          <button type="button" class="close" data-dismiss="modal">&times;</button>
          <h4 class="modal-title">Featured Product Setting</h4>
        </div>
        <div class="modal-body">
            <div class="form-group row">
                <label class="col-md-2 form-control-label" for="start_date">Start date</label>
                <div class="col-md-3">
                    <input type="date" id="start_date" name="start_date" value="" required>
                </div>
            </div>
            <div class="form-group row">
                <label class="col-md-2 form-control-label" for="end_date">End date</label>
                <div class="col-md-3">
                    <input type="date" id="end_date" name="end_date" value="">
                </div>
            </div>
        
        </div>
        <div class="modal-footer">
            <button type="button" class="btn btn-primary" id="ok" data-dismiss="modal">Ok</button>
            <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
          
        </div>
      </div>
      
    </div>
  </div>
  
</div>
<script>
  $(document).off('click','#ok');
  $(document).on('click','#ok', function(){
      console.log($('#start_date').val());
  })
</script>
</body>
</html>
`
